outerbase-install.sh

```diff
@@ -259,59 +249,135 @@ fi
 ### outer filesystem
 ###
 
-mkdir /mnt/outer
+OUTER_CHROOT=$INNER_CHROOT/outer
+mkdir $OUTER_CHROOT
 
 if [ -z "$customdrives" ]; then
   outerbasedevice=/dev/gpt/outer
 fi
 
 newfs -m2 $outerbasedevice
-mount $outerbasedevice /mnt/outer
+mount $outerbasedevice $OUTER_CHROOT
 
 
 ###
-### outer base install
+### bootstrap pkg
 ###
 
-# use custom outerbase.txz if set
-if [ -z "$outerbasetxz" ]; then
-  outerbasetxz=/usr/freebsd-dist/base.txz
+# Bootstrap pkg if not already available
+if ! pkg -N > /dev/null 2>&1; then
+  echo "Bootstrapping pkg on the host system"
+  pkg bootstrap -y
 fi
 
-# extract /var but leave it empty for varmfs
+
+###
+### pkgbase installation function
+###
+
+# Install FreeBSD base system via pkgbase packages
+# Args: $1 = chroot path, $2 = description (e.g., "outer"), $3 = space-separated package sets
+install_pkgbase() {
+  local chroot_path="$1"
+  local desc="$2"; shift 2
+  local user_pkgsets="$*"
+
+  echo "Installing $desc base system via pkgbase..."
+
+  local repos_dir=/usr/share/bsdinstall
+  local pkg_cmd="pkg --rootdir $chroot_path --repo-conf-dir $repos_dir -o IGNORE_OSVERSION=yes"
+
+  # Copy pkg keys to chroot
+  mkdir -p "$chroot_path/usr/share/keys"
+  cp -R /usr/share/keys/* "$chroot_path/usr/share/keys/"
+
+  # Update pkg repositories
+  $pkg_cmd update
+
+  # Build package list - always install minimal, kernel, and pkg (if available)
+  local packages="FreeBSD-set-minimal FreeBSD-kernel-generic"
```

Comment:
I think we want `FreeBSD-kernel-generic` only for outer. Will fix that when I get a chance.